src/manager/event_handlers.ts

```diff
@@ -385,6 +386,7 @@ function refreshShadow(actor: ExtensionsWindowActor) {
     const {border_radius, padding} = getRoundedCornersCfg(win);
 
     updateShadowActorStyle(win, shadow, border_radius, shadowSettings, padding);
+    refreshRoundedCorners(actor);
```

Comment:
Why are you refreshing rounded corners in `refreshShadow`? Since this only has to be called when window focus is changed, you should instead change the definition of `onFocusChanged` to something like this:

```typescript
export function onFocusChanged(actor: ExtensionsWindowActor) {
  refreshShadow(actor);
  refreshRoundedCorners(actor);
}
```

It's also possible to only update the border color uniform without recalculating everything else (see implementation of `RoundedCornersEffect.update_uniforms` for details), but I think that would just add a bunch of unnecessary functions and abstraction level mixing. Though if you think it may be a good option you can still try and see if you like how it feels.